### ISSUE TITLE
Distribute RoundhousE on Docker?

### ISSUE THREAD

**Issue by danrien** [chucknorris/roundhouse#342]
Hi all,

Would it be possible to officially distribute a version of RoundhousE in a .Net core based docker container? I've whipped together a quick version of RoundhousE running in a docker container, but it's probably not the best way to do it:

https://github.com/namehillsoftware/dotnet-core-roundhouse/blob/master/Dockerfile

**Comment by erikbra** [comment]
Interesting option!

What would be the scenarios where this would be better than e.g. installing RoundhousE as a .net core global tool? Would it be in scenarios where .net core is not installed on the build server, but docker is installed?

**Comment by danrien** [comment]
Beyond some of the basic benefits of Docker (repeatable installations and deployment, isolation from other processes), I've found it most useful if you want to use RoundhousE within a larger docker deployment, for example, within a docker compose:

```
  sql-server:
    hostname: sql-server
    image: microsoft/mssql-server-linux:2017-CU8
    environment:
      - "ACCEPT_EULA=Y"
      - "SA_PASSWORD=strong(!)Password"
    ports:
      - "${DB_PORT:-2433}:1433"
  roundhouse-deploy:
    image: namehillsoftware/dotnet-core-roundhouse
    depends_on:
      - sql-server
    volumes:
      - "${ROUNDHOUSE_SQL_SCRIPTS_DIR}:/scripts"
    command: [ "-c", "Server=sql-server;Database=MyDb;User Id=SA;Password=strong(!)Password;Connect Timeout=120", "-o", "/output" ]
```

**Comment by BiggerNoise** [comment]
Not a bad suggestion. I think we can also set up our docker instance to watch the repo.

If that's the case, this will probably happen, eventually.

If not, I don't know if I want to add yet another place we need to remember to release to.

**Comment by erikbra** [comment]
I think this is a cool idea. I am not awfully familiar with docker (using it for some development, e.g. running sql server on macOS), but nothing in production. So, what would we have to do, @danrien ? Just add a Dockerfile to some public docker registry? I think yours looks like a good start, however, I think I would use the dotnet global tool deployment, now that it's available (in release/1.0.0 branch, that is).

**Comment by danrien** [comment]
@erikbra you just sign up on docker and link a Github repository with dockerhub, and all of your branches show up. For example, I created an image today based off of the RoundhousE dotnet tool, and you can see it appeared here, alongside "latest", which is off master:

https://hub.docker.com/r/namehillsoftware/dotnet-core-roundhouse/tags/
